roles/v2/mining-proxy/src/lib/template_receiver/mod.rs

```diff
@@ -64,7 +67,9 @@ impl TemplateRx {
         .try_into()
         .unwrap();
         Self::send(self_mutex.clone(), sv2_frame).await;
+        let cloned = self_mutex.clone();
 
+        tokio::task::spawn(async { Self::on_new_solution(cloned, solution_receiver).await });
```

Comment:
Looks like this could be simplified to `tokio::task::spawn(Self::on_new_solution(cloned, solution_receiver));`.